compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/ModuleLevelVariableDeclarationAnalysisTask.java

```diff
@@ -0,0 +1,100 @@
+import io.ballerina.projects.DocumentId;
+import io.ballerina.projects.plugins.AnalysisTask;
+import io.ballerina.projects.plugins.SyntaxNodeAnalysisContext;
+import io.ballerina.stdlib.graphql.compiler.schema.generator.GraphqlModifierContext;
+import io.ballerina.stdlib.graphql.compiler.schema.generator.SchemaGenerator;
+import io.ballerina.stdlib.graphql.compiler.schema.types.Schema;
+import io.ballerina.stdlib.graphql.compiler.service.InterfaceFinder;
+import io.ballerina.stdlib.graphql.compiler.service.validator.ServiceValidator;
+
+import java.util.Map;
+
+import static io.ballerina.stdlib.graphql.compiler.Utils.hasCompilationErrors;
+import static io.ballerina.stdlib.graphql.compiler.Utils.isGraphQLServiceObjectDeclaration;
+import static io.ballerina.stdlib.graphql.compiler.schema.generator.GeneratorUtils.getDescription;
+
+/**
+ * Validates a Ballerina GraphQL Service variable declaration.
+ */
+public class ModuleLevelVariableDeclarationAnalysisTask implements AnalysisTask<SyntaxNodeAnalysisContext> {
+    private final Map<DocumentId, GraphqlModifierContext> modifierContextMap;
+
+    public ModuleLevelVariableDeclarationAnalysisTask(Map<DocumentId, GraphqlModifierContext> nodeMap) {
+        this.modifierContextMap = nodeMap;
+    }
+
+    @Override
+    public void perform(SyntaxNodeAnalysisContext context) {
+        if (hasCompilationErrors(context)) {
+            return;
+        }
+
+        ModuleVariableDeclarationNode moduleVariableDeclarationNode = (ModuleVariableDeclarationNode) context.node();
+        if (!isGraphQLServiceObjectDeclaration(moduleVariableDeclarationNode)) {
+            return;
+        }
+
+        if (moduleVariableDeclarationNode.initializer().isEmpty()) {
+            return;
+        }
+        ObjectConstructorExpressionNode graphqlServiceObjectNode
+                = (ObjectConstructorExpressionNode) moduleVariableDeclarationNode.initializer().get();
+        InterfaceFinder interfaceFinder = new InterfaceFinder();
+        interfaceFinder.populateInterfaces(context);
+        ServiceValidator serviceObjectValidator = new ServiceValidator(context, graphqlServiceObjectNode,
+                interfaceFinder);
+        serviceObjectValidator.validate();
+        if (serviceObjectValidator.isErrorOccurred()) {
+            return;
+        }
+        DocumentId documentId = context.documentId();
+        String description = null;
+        if (context.semanticModel().symbol(moduleVariableDeclarationNode).isPresent()) {
+            VariableSymbol serviceVariableSymbol = (VariableSymbol) context.semanticModel()
+                    .symbol(moduleVariableDeclarationNode).get();
+            description = getDescription(serviceVariableSymbol);
+        }
+        Schema schema = generateSchema(interfaceFinder, graphqlServiceObjectNode, description, context);
+        addToModifierContextMap(documentId, moduleVariableDeclarationNode, schema);
```

Comment:
These seem to be similar to the other analysis task code. Can we use a single method to call from both places instead of duplicating the code?

Reply:
Common implementations are moved to an abstract class